File: Difficulty: Easy/Special Stack/special-stack.py
# Your task is to complete all these function's
# function should append an element on to the stack
def push(arr, ele):
    arr.append(ele)

# Function should pop an element from stack
def pop(arr):
    if arr:
        return arr.pop()

# function should return 1/0 or True/False
def isFull(n, arr):
    return len(arr)>=n

# function should return 1/0 or True/False
def isEmpty(arr):
    return len(arr)==0

# function should return minimum element from the stack
def getMin(n, arr):
    if arr:
        return min(arr)
    return None

File: Difficulty: Easy/Special Stack/special-stack.js
class SpecialStack {
    constructor(capacity = Infinity) {
        this.stack = [];
        this.minStack = [];
        this.capacity = capacity;
    }

    /**
     * @param {number} x
     * @return {void}
     */
    push(x) {
        if (this.stack.length < this.capacity) {
            this.stack.push(x);
            if (this.minStack.length === 0) {
                this.minStack.push(x);
            } else {
                this.minStack.push(Math.min(x, this.minStack[this.minStack.length - 1]));
            }
        }
    }

    /**
     * @return {number}
     */
    pop() {
        if (this.stack.length === 0) return -1;
        this.minStack.pop();
        return this.stack.pop();
    }

    /**
     * @return {boolean}
     */
    isFull() {
        return this.stack.length >= this.capacity;
    }

    /**
     * @return {boolean}
     */
    isEmpty() {
        return this.stack.length === 0;
    }

    /**
     * @return {number}
     */
    getMin() {
        if (this.minStack.length === 0) return -1;
        return this.minStack[this.minStack.length - 1];
    }
}